DBZ-9319: Update Platform doc on Auto format and Using DBZ server config to create pipeline (#6715)
debezium/debezium: documentation/modules/ROOT/pages/operations/debezium-platform.adoc

You can use the **Smart Editor** to specify the JSON that defines the source configuration.
You can enter and edit `JSON` directly in the editor, or paste `JSON` from an external source into the editor.
With a few small differences, the JSON that you use to configure a data source in the {prodname} management Platform is nearly identical to the JSON in the `config` section that defines the configuration for a {prodname} connector on Kafka Connect or in the {prodname} Server.

Now stage UI provides you the ability to directly use the configuration of a {prodname} connector on Kafka Connect or in {prodname} Server configuration by either uploading or pasting into the **Smart Editor**.

> Stage UI will intelligently recognize the configuration type and will provide the appropriate prompt in the UI along with **Auto format** action button in the **Smart Editor** toolbar to automatically convert provided JSON into the required JSON format for the platform.  

Reviewer: ```suggestion
Stage UI will intelligently recognize the configuration format and will provide the appropriate prompt in the UI along with **Auto format** action button in the **Smart Editor** toolbar to automatically convert provided JSON into the required JSON format for the platform.  
```

**Form Editor**:: Enables you to specify the name and description of the destination, along with a list of properties.
For a complete list of the properties that are available for a sink connector, see the connector documentation.
Smart Editor:: Enables you to define the sink configuration in JSON. 
Alternately, you can extract a {prodname} Server sink configuration and automatically reformat it for use in the platform.

[NOTE]
====
image::create-destination-toolbar.png[{prodname} platform - Create sink toolbar]

> To create a sink destination from an existing {prodname} Server configuration, from the toolbar on the *Destination catalog* page, click *Create using smart editor* .

Reviewer 2: Ah, ok, I see, I just found `create-destination-toolbar.png`.

In that case, I think it's unnecessary to refer to the toolbar, because the button itself is obvious. I've adjusted my comment accordingly. Thanks!

debezium.sink.type=pubsub
debezium.sink.pubsub.project.id=debezium-tutorial-local
debezium.sink.pubsub.address=pubsub:8085
debezium.source.connector.class=io.debezium.connector.mysql.MySqlConnector
debezium.source.database.hostname=mysql
debezium.source.database.server.id=223344
debezium.source.database.port=3306
debezium.source.database.user=debezium
debezium.source.database.password=dbz
debezium.source.schema.history.internal=io.debezium.storage.file.history.FileSchemaHistory
debezium.source.schema.history.internal.file.filename=data/schema.dat
debezium.source.offset.storage.file.filename=data/offsets.dat
debezium.source.offset.flush.interval.ms=0
debezium.source.topic.prefix=tutorial
debezium.source.database.include.list=inventory
debezium.source.table.include.list=inventory.customers

# ..
----

When you add this properties file to the Smart Editor, it intelligently recognizes that the provided configuration is a Debezium Server configuration and will inform via displaying an alert.

image::debezium-smart-editor-server.png[{prodname} UI smart editor - Server]

> The Smart Editor toolbar will then display an **Auto Format** option (highlighted in the image above). You can use this option to automatically extract the sink configuration and convert the sink properties into the required Debezium Platform JSON format.

Reviewer 2: ```suggestion
The Smart Editor toolbar displays an **Auto Format** option, as shown in the preceding image. 
Select this option to automatically extract the sink configuration and convert the sink properties into the JSON format the {prodname} Platform supports.
```

> The following image shows the auto formatted JSON that you would use to define a Pub-Sub sink where you have to just provide name and optionally add some appropriate description in the JSON.

Reviewer 2: ```suggestion
The following image shows automatically formatted JSON that defines a publish-subscribe sink.
Edit the JSON to update the value of the `name` field, and optionally add populate the `description` field.
```

Author: feels like something is missing here, "_and optionally add populate the `description` field._" ?

image::debezium-platform-smart-editor-sink.png[{prodname} platform smart editor - Sink]

> Final JSON after updating the name and description

Reviewer 2: ```suggestion
After you update the name and description, the following JSON results:
```

To edit a destination, go to the `Destination` menu and then click the `action` menu of the destination you want to edit, then click `Edit`.

[NOTE]

> Editing a destination will affect all pipelines that use it. A confirmation modal will also appear informing this before changes are being submitted from the UI.

Reviewer 2: ```suggestion
Editing a destination affects all pipelines that use it.
After you make a change, a confirmation message informs you that pipelines that use the destination will restart.
```

[.responsive]
video::cY7TS0nWBoY[youtube, title="Create, edit and remove a pipeline"]

> ==== Creating pipeline

Reviewer 2: Use plural nouns with gerund phrases when possible. This should either be _Creating pipelines_, or _Creating a pipeline_.

> From the platform UI, open the **Pipeline** menu and click **Create your first pipeline** to open the **Pipeline Designer**.

Reviewer 2: ```suggestion
From the platform UI, open the *Pipeline* menu, and then click *Create your first pipeline*.
The *Pipeline Designer* opens.
```

From the Pipeline Designer you can add the pieces for your data pipeline.

> . Click the **+ Source** box to add a source, and then choose a previously created source, or create a new source.

Reviewer 2: The double asterisks do work, but a pair of single asterisks is sufficient to apply bold formatting.

. (Optional) Click the **+ Transform** box to apply one or more transformations.

. Click the **+ Destination** box to add a destination,and then choose a previously created destination, or create a new destination.

> The order in which a Transform is applied to the source is important and decides the final output at the end, so you have to be mindfully of the order in which you add transforms.

Reviewer 2: ```suggestion
The sequence in which {prodname} applies transformations to the source is significant and affects the final output. 
Be sure to specify a sequence that results in the expected output. 
```

> In the pipeline designer you have the ability to delete a transform or alter the sequency of applied transforms. For more information about using the pipeline designer, see xref:debezium-platform-using-the-pipeline-designer-to-remove-and-order-transformations[using the Pipeline designer to remove and order transformations]

Reviewer 2: ```suggestion
In the pipeline designer you can delete a transform, or alter the sequence of applied transforms. 
For more information about using the pipeline designer, see xref:debezium-platform-using-the-pipeline-designer-to-remove-and-order-transformations[using the Pipeline designer to remove and order transformations].
```

[NOTE]
Transformations that are configured with a predicate are marked with a predicate icon (image:predicate-icon.png[Transformation configured with a predicate]).
A tooltip shows the name of the predicate type.

image::debezium-platform-pipeline.png[{prodname} platform pipeline]

After you finish designing your pipeline, click **Configure Pipeline**, and then specify the name, description, and logging level for the pipeline.

> ==== Creating pipeline using {prodname} Server config

Reviewer 2: ```suggestion
==== Creating a pipeline from a {prodname} Server configuration
```

> Pipeline Designer also provides you the option to upload the {prodname} Server configuration properties file directly to create the data pipeline resources i.e source, destination and transforms.

Reviewer 2: ```suggestion
The Pipeline Designer provides an option to create data pipeline resources (that is, source, destination, and transforms) by uploading an existing {prodname} Server configuration properties file .
```

> In the pipeline designer click on **DBZ server config** which will open a modal where you can upload or drag and drop your {prodname} server properties file. Once you upload the file click on **Create** button.

Reviewer 2: ```
*DBZ server config*
```
Is that correct? Or should it read *{prodname} server config(uration)*?  I don't see a PNG file that shows the actual UI.

> In the pipeline designer click on **DBZ server config** which will open a modal where you can upload or drag and drop your {prodname} server properties file. Once you upload the file click on **Create** button.

Reviewer 2: ```suggestion
* In the pipeline designer, click *DBZ server config*, upload, or drag and drop a {prodname} server properties file, and then click *Create*.
```

> This will automatically create Source, Transforms and Destination in synchronous manner using the {prodname} server configuration. 

Reviewer 2: ```suggestion
The designer creates the Source, Transforms, and Destination for the pipeline, based on the {prodname} server configuration that you provided. 
```

image::debezium-platform-pipeline-server-modal.png[{prodname} platform pipeline server modal]

[NOTE]

> Source and destination names are autogenerated.

Reviewer 2: ```suggestion
The source and destination names are generated automatically.
```

==== Deleting a pipeline

From the platform UI, open the **Pipeline** menu, click the **Action** menu for the pipeline that you want to delete, and then click **Delete**.
The deletion removes only the pipeline: the source, destination, and any transformations are not deleted.

==== Editing a pipeline

. From the platform UI, open the **Pipeline** menu, click the **Action** menu for the pipeline that you want to edit, and then click **Edit**.

> . Use the pipeline designer to modify transformations as needed. For more information about using the pipeline designer, see xref:debezium-platform-using-the-pipeline-designer-to-remove-and-order-transformations[Using the Pipeline designer to remove and order transformations]

Reviewer 2: ```suggestion
. In the pipeline designer, modify transformations as needed. 
For more information about using the pipeline designer, see xref:debezium-platform-using-the-pipeline-designer-to-remove-and-order-transformations[Using the Pipeline designer to remove and order transformations].
```

> . Click on **Save and next** to edit the pipeline name, description, and log level properties of the pipeline.

Reviewer 2: ```suggestion
. Click *Save and next* to edit the name, description, and log level properties of the pipeline.
```
